File: esprit-internship2-back/src/main/java/tn/esprit/espritinternship2back/entity/Personnel.java
package tn.esprit.espritinternship2back.entity;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.Date;
import java.util.List;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Personnel {

    @Id
    @GeneratedValue
    private Long id;
    private String firstName = "";
    private String lastName = "";
    private int phone;
    private String email = "";
    private Date birthDate;
    @Lob
    @Column(columnDefinition = "LONGTEXT")
    private String photo = "";
    private Float salary;

    @Enumerated(EnumType.STRING)
    private PersonnelRole personnelRole;

    private int state = 0;

    public enum PersonnelRole {
        Client,
        Admin,
        Driver,
        Worker,
        Mechanic,
        Technician,
        Monitor
    }
}

File: esprit-internship2-front/src/models/Personnel.ts
export interface Personnel {
  id?: number;
  firstName?: string;
  lastName?: string;
  password?: string;
  phone?: number;
  email?: string;
  birthDate?: string; 
  photo?: string;
  salary?: number;

  personnelRole?: PersonnelRole;
  
  state?: number;
}

export enum PersonnelRole {
  Client = 'Client',
  Admin = 'Admin',
  Driver = 'Driver',
  Worker = 'Worker',
  Mechanic = 'Mechanic',
  Technician = 'Technician',
  Monitor = 'Monitor'
}
